File: lib/db/schema.ts
import { sqliteTable, text, integer } from 'drizzle-orm/sqlite-core';

export const sessions = sqliteTable('sessions', {
  id: text('id').primaryKey(),
  createdAt: text('created_at').notNull(),

  // Checkin data (stored flat for easy querying)
  checkinMood: integer('checkin_mood').notNull(),
  checkinTension: integer('checkin_tension').notNull(),
  checkinSelfCritical: integer('checkin_self_critical', { mode: 'boolean' }).notNull(),
  checkinIntent: text('checkin_intent').notNull(),
  checkinLastOutcome: text('checkin_last_outcome'),
  checkinFreeText: text('checkin_free_text'),

  // Supervisor decision (JSON fields for arrays)
  riskLevel: text('risk_level').notNull(),
  patterns: text('patterns').notNull(),         // JSON: HarmfulPattern[]
  action: text('action').notNull(),
  recommendedMode: text('recommended_mode').notNull(),
  supervisorMessage: text('supervisor_message').notNull(),
  guidanceDuration: integer('guidance_duration').notNull(),

  // Guidance used
  guidanceMode: text('guidance_mode').notNull(),
  guidanceText: text('guidance_text').notNull(),
  guidanceIsPreset: integer('guidance_is_preset', { mode: 'boolean' }).notNull(),

  // Post-session (nullable, updated via PATCH)
  postFeltBetter: integer('post_felt_better', { mode: 'boolean' }),
  postWouldContinue: integer('post_would_continue', { mode: 'boolean' }),
  postNotes: text('post_notes'),

  // Summary (generated after post-reflection)
  summary: text('summary'),
});

File: komorebi-server/src/main/scala/komorebi/db/Schema.scala
package komorebi.db

import doobie.*
import doobie.implicits.*
import cats.effect.IO

object Schema:

  val createSessionsTable: Update0 = sql"""
    CREATE TABLE IF NOT EXISTS sessions (
      id TEXT PRIMARY KEY,
      created_at TEXT NOT NULL,
      checkin_mood INTEGER NOT NULL,
      checkin_tension INTEGER NOT NULL,
      checkin_self_critical INTEGER NOT NULL,
      checkin_intent TEXT NOT NULL,
      checkin_last_outcome TEXT,
      checkin_free_text TEXT,
      risk_level TEXT NOT NULL,
      patterns TEXT NOT NULL,
      action TEXT NOT NULL,
      recommended_mode TEXT NOT NULL,
      supervisor_message TEXT NOT NULL,
      guidance_duration INTEGER NOT NULL,
      guidance_mode TEXT NOT NULL,
      guidance_text TEXT NOT NULL,
      guidance_is_preset INTEGER NOT NULL,
      post_felt_better INTEGER,
      post_would_continue INTEGER,
      post_notes TEXT,
      summary TEXT,
      reflection_profile TEXT,
      reflection_summary TEXT
    )
  """.update

  val createUserMemoryTable: Update0 = sql"""
    CREATE TABLE IF NOT EXISTS user_memory (
      user_id TEXT PRIMARY KEY,
      memory TEXT NOT NULL,
      updated_at TEXT NOT NULL
    )
  """.update

  def initialize(xa: Transactor[IO]): IO[Unit] =
    val program = for
      _ <- createSessionsTable.run
      _ <- createUserMemoryTable.run
    yield ()
    program.transact(xa)
